README.md

```diff
@@ -38,45 +42,52 @@ You can access DOM elements with javascript.
 
 ```html
-<li class="menu-item">
-  London
-</li>
-<li class="menu-item">
-  Nazareth
+<li class="menu-item">London
 </li>
+<li class="menu-item">Nazareth</li>
 <script>
   var menuItems = document.getElementsByClassName("menu-item");
-  // [_li.menu-item_,_li.menu-item_]
+  // [li.menu-item, li.menu-item]
 </script>
 ```
 
-`document.querySelector(myCssSelector)` returns the first element matching `myCssSelector`, where `myCssSelector` takes the form of a [CSS selector](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Selectors) for example "#myId", ".myClass", "myTag", etc etc.
+It's important to note that this collection is _live_. If you change one of the elements within it, then try to access the collection again it will be updated. This is different to how most function return values work, so be careful.
+
+`document.querySelector(myCssSelector)` returns the first element matching `myCssSelector`, where `myCssSelector` is any valid [CSS selector](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Selectors). For example "#myId", ".myClass", "myTag", etc etc.
+
+```js
+document.querySelector("#myId"); // logs first element with ID of "myId"
+document.querySelector(".myClass"); // logs first element with class name of "myClass"
+document.querySelector("p > a.active"); // logs first anchor element with class name of "active" that is within a paragraph
+```
+
+`document.querySelectorAll(myCssSelector)` returns a static [NodeList](https://developer.mozilla.org/en-US/docs/Web/API/NodeList) of all elements matching myCssSelector.
+
 ```html
-<li class="menu-item">
-  London
-</li>
-<li class="menu-item">
-  Nazareth
-</li>
+<button id="btn-1">Sign up</button>
+<button id="btn-2">Sign in</button>
 <script>
-  var firstMenuItem = document.querySelector(".menu-item");
-  // <li class="menu-item">London</li>
+  var buttons = document.querySelectorAll("button");
+  // [button#btn-1, button#btn-2];
 </script>
 ```
-`document.querySelectorAll(myCssSelector)` returns an _array-like object_ of all elements matching myCssSelector.
+It's important to note that this NodeList is _not_ live. Whatever `querySelectorAll` returns will stay the same even if the underlying elements later change. Think of it as a snapshot of the DOM at the time you ran the function.
+
+NodeLists are similar to arrays, but with fewer properties. For example they have a `.forEach()` method, but no `.map()` or `.filter()`. If you need to do more complex operations you can turn them into arrays using `Array.from(myNodeList)`.
+
 ```html
-<li class="menu-item">
-  London
-</li>
-<li class="menu-item">
-  Nazareth
-</li>
+<button id="btn-1">Sign up</button>
+<button id="btn-2">Sign in</button>
 <script>
-  var firstMenuItem = document.querySelectorAll(".menu-item");
-  // [_li.menu-item_,_li.menu-item_];
+  var buttons = document.querySelectorAll("button");
+  buttons.forEach(button => {
+    console.log(button);
+  }
```

Comment:
Missing closing bracket. Should be `});`

Reply:
Good catch!
```suggestion
  })'
```